### ISSUE TITLE
Polka/Sapper example with uws

### ISSUE THREAD

**Issue by ansarizafar** [lukeed/polka#24]
Is it possible to use polka/sapper with uws?

**Comment by lukeed** [comment]
Hey, Polka & Sapper are two separate projects. The example was just to show how to use them together.

But, yes, you can use `uws` with Polka (and Sapper). There's already [an example with `uws`'s HTTP module](https://github.com/lukeed/polka/tree/master/examples/with-uws), but you can definitely attach the socket server too. I have an example for this in the works.

**Comment by ansarizafar** [comment]
> There's already an example with uws's HTTP module

This example is not using Sapper with Polka and uws and I am unable to understand how Polka, uws and Sapper can be used together. Could you please provide an example of using Polka, uws and  Sapper together.

**Comment by Youngestdev** [comment]
There's an example already
 Check the examples folder and you'll find it.

**Comment by ansarizafar** [comment]
@Youngestdev There is no such example in examples folder. I am talking about Polka + uws + Sapper example.

**Comment by Youngestdev** [comment]
@lukeed

**Comment by lukeed** [comment]
@ansarizafar I actually answered you in Svelte Gitter since you pinged me there. Here's the rough layout, which you could totally piece together from the two examples:

```js
const polka = require('polka');
const { Server } = require('uws');
const sapper = require('sapper')();
const static = require('serve-static')('assets');
const compression = require('compression')({ threshold:0 });

const { PORT=3000 } = process.env;

// (SAPPER) this allows us to do e.g. `fetch('/api/blog')` on the server
const fetch = require('node-fetch');
global.fetch = (url, opts) => {
	if (url[0] === '/') url = `http://localhost:${PORT}${url}`;
	return fetch(url, opts);
};

// Init Polka with Sapper, Compression, and Static Assets middlewares
// grab the `server` key, which is HTTP server
const { server } = polka().use(compression, static, sapper);

// Init uWS Server, using HTTP server
const wss = new Server({ server });

// Do whatever you want with sockets
// You have to connect to them on client-side
wss.on('connection', ws => {
  ws.on('message', msg => {
    console.log('received: %s', msg);
  });
  ws.send('something');
});

// Start listening to HTTP server
server.listen(PORT, _ => {
  console.log(`> Ready on localhost:${PORT}`);
});
```

For client-side socket listener, you can use native `WebSocket` or my [`sockette`](https://github.com/lukeed/sockette) library.

You have more than enough to finish from here. All this information is easily found in Sapper's, Polka's, and(u)WS's documentations. Just need to look a little bit.

@Youngestdev I already get notifications for everything, no need to ping 😆

**Comment by Youngestdev** [comment]
Okay !

**Comment by ansarizafar** [comment]
Thanks for the exmaple but I think I was not able to explain my question clearly. I wanted to know, Is it possible to use uws.http server with Polka router/middleware and Sapper. If I understood it correctly, In the above example, we are only using uws socket server to handle websocket request. What I want to achieve is to handle all http requests via uws.http.

**Comment by lukeed** [comment]
Yes. The example becomes even simpler and you can use the `with-uws` example directly.

**Comment by ansarizafar** [comment]
uws example in this repo only shows how to handle a single route. My question is how can we handle all sapper routes via uws.http server.

**Comment by lukeed** [comment]
It's the exact same thing man, you'll get it.

**Comment by ansarizafar** [comment]
I don't think its the same thing. I am not an expert, but in my opinion It will only be possible If we somehow can replace polka's http server with uws.http server.  Anyways thanks for the reply.

**Comment by lukeed** [comment]
Look at https://github.com/lukeed/polka/blob/master/examples/with-https/index.js

Define your Polka app, with routing.

Create the UWS HTTP server, passing Polka's `handler` to it.

You can do this for _any_ custom HTTP server, that's what both the UWS & HTTPS examples are showing.

**Comment by ansarizafar** [comment]
https example shows a single handler for all routes. and uws example shows how to pass that handler to uws.http server. The handler will always return POLKA: Hello from ${req.pathname} for all routes. What about Sapper routes? How sapper routes will work with this code

```
polka().use(compression, static, sapper);

const { handler } = polka().get('*', (req, res) => {
	res.end(`POLKA: Hello from ${req.pathname}`);
});

http.createServer(handler).listen(PORT, err => {
	console.log(`> Running on localhost:${PORT}`);
});

````

**Comment by ansarizafar** [comment]
As I mentioned before I am not an expert but I have used express in the past and now a days I am using my own fastify like framework build on top of native http and router module and graphql inspired data layer with single endpoint and request batching support and my own front end framework build on top of hyperhtml with global store mangement via javascript proxies. I have tried to use uws.http many times in the past without any success because (as you correctly mentioned) UWS's http is intentionally incomplete & there are probably missing methods or bugs. After I saw uws.http example I was wondering How its working with polka and sapper. Thanks for complete example and details explanation. I also Thank you for Polka and all your opensource work.
